### PR TITLE
feat(Tag): Tag component

### PR DESCRIPTION
**Pull request by reshav-abraham**
closes https://github.com/narmi/design_system/issues/344

Initial tag component.

 Referenced spec https://zeroheight.com/8ac87d4ba/p/08acdf-tags

### CODE REVIEW

File: src/Tag/index.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    color: RGB(var(--rgb-white));
+    cursor: pointer;
+    padding-left: 8px;
+    font-size: 12px;
+    position: relative;
+    top: 1px;
+  }
+}
+.nds-tag--dismissible {
+  color: RGB(var(--rgb-white));
```
Comment (akdetrick) [inline src/Tag/index.scss:17]: the `--color` variables don't need the RGB wrapper - you can just do:

```css
color: var(--color-white);
```

The `--rgb` variables are for when you need to transparentize a color by mixin in an alpha channel like `rgba(var(--rgb-pine), var(--alpha-20));`

File: src/Tag/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+ const noop = () => {};
+
+const Tag = ({
+  kind = "subdued", // outline, subdued, x-tag (cactus400) #7fbc5b; #7FBC5B oneof
+  onDismiss = noop,
+  label = "my cool tag",
+}) => {
+  return (
+    // <AsElement
```
Comment (akdetrick) [inline src/Tag/index.js:27]: ✂️ we could take out the `AsElement` support for now

File: src/Tag/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { useLayer } from "react-laag";
+import cc from "classcat";
+import AsElement from "../util/AsElement";
+
+/**
```
Comment (akdetrick) [inline src/Tag/index.js:7]: ✂️ If you update this big jsDoc comment and put it directly above the `const Tag` function, the docs will show up in storybook above the "overview" story

File: src/Tag/index.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { useLayer } from "react-laag";
+import cc from "classcat";
+import AsElement from "../util/AsElement";
```
Comment (akdetrick) [inline src/Tag/index.js:5]: don't need `AsElement` for now

File: src/Tag/index.scss [hunk 5]
```diff
@@ -0,0 +1,30 @@
+.nds-tag {
+  padding: 8px;
+  border-radius: 16px;
+  display: flex;
+  align-items: baseline;
+  font-size: 12px;
```
Comment (akdetrick) [inline src/Tag/index.scss:6]:
```suggestion
  font-size: var(--font-size-xs);
```

File: src/Tag/index.scss [hunk 6]
```diff
@@ -0,0 +1,30 @@
+.nds-tag {
+  padding: 8px;
```
Comment (akdetrick) [inline src/Tag/index.scss:2]: you could do one of two things here:

use the var:
```css
padding: var(--space-s);
```

or apply the padding via a helper class

```jsx
{/* padding--all--s applies `s` padding (8px) to all sides */}
<div className={`nds-typography nds-tag nds-tag--${kind} padding--all--s`}>
```

**Review by akdetrick — CHANGES_REQUESTED** [review]
Left some comments; looks like this also needs a rebase from master

File: src/Tag/index.js (hunk 3 above)
Comment (reshav-abraham) [reply src/Tag/index.js:7]: Wrote a new description. Will defer to any suggestions you have!

File: src/Tag/context.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { createContext } from "react";
```
Comment (akdetrick) [inline src/Tag/context.js:1]: Looks like this got copied over by mistake ✂️

File: src/Tag/index.scss [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    padding-left: 8px;
+    font-size:var(--font-size-xs);
+    position: relative;
+    top: 1px;
+  }
+}
+.nds-tag--dismissible {
+  color: var(--color-white);
+  background: var(--color-cactus);
+  border: 1px solid var(--rgb-cactus);
```
Comment (akdetrick) [inline src/Tag/index.scss:19]:
```suggestion
  border: 1px solid var(--color-cactus);
```

**Review by akdetrick — CHANGES_REQUESTED** [review]
Just some minor things, but otherwise looks good

File: tokens/src/color/narmi.json [hunk 9]
```diff
@@ -75,6 +75,7 @@
       "grey": { "value": "#4C4C4C" },
       "mediumGrey": { "value": "#8C8C8C" },
       "lightGrey": { "value": "#D9D9D9" },
+      "smokeGrey": { "value": "#F3F3F3" },
```
Comment (akdetrick) [inline tokens/src/color/narmi.json:78]: The token `--bgColor-smokeGrey` already exists, but I am now seeing that it's confusing to use a `color` prefix for some tokens and a `bgColor` prefix for others.

I might copy more of those `bgColor` tokens into the `color` category...

**Review by akdetrick — CHANGES_REQUESTED** [review]
a few minor things, but otherwise looks good

**Review by akdetrick — APPROVED** [review]

File: src/Tag/index.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+const noop = () => {};
+
+/**
+ * A rounded rectangle inline label.
+ * The user has the option of firing a callback for 'dismissible' Tags.  
+ */
+const Tag = ({
+  kind = "subdued", // outline, subdued, x-tag (cactus400) #7fbc5b; #7FBC5B oneof
+  onDismiss = noop,
+  label = "my cool tag",
```
Comment (akdetrick) [inline src/Tag/index.js:14]: might be good to make this an empty string by default - if we're missing data or something else makes the `label` undefined, this message could show up in prod

Comment (reshav-abraham) [reply src/Tag/index.js:14]: Good call!

File: tokens/src/color/narmi.json (hunk 9 above)
Comment (reshav-abraham) [reply tokens/src/color/narmi.json]: This line actually introduces this warning  while building
```
cssRgbColors
⚠️ /Users/reshavabraham/work/narmi/design_system/dist/tokens/css/rgbColors.css
While building rgbColors.css, token collisions were found; output may be unexpected.
    Output name rgb-smokeGrey was generated by:
        color.background.smokeGrey   243, 243, 243
        color.narmi.smokeGrey   243, 243, 243
This many-to-one issue is usually caused by some combination of:
    * conflicting or similar paths/names in property definitions
    * platform transforms/transformGroups affecting names, especially when removing specificity
    * overly inclusive file filters
```
So I've gone with `--bgColor-smokeGrey`

File: src/Tag/index.scss [hunk 11]
```diff
@@ -0,0 +1,30 @@
+.nds-tag {
+  padding:var(--space-xs);
+  border-radius: 16px;
+  display: flex;
+  align-items: baseline;
+  font-size: var(--font-size-xs);
+  .narmi-icon-x {
+    color: var(--color-white);
+    cursor: pointer;
+    margin-left: 8px;
```
Comment (reshav-abraham) [inline src/Tag/index.scss:10]: This needs to be margin instead of padding because the padding space makes the clickable area bigger

Comment (akdetrick) [reply src/Tag/index.scss:10]: good catch!

**Pull request merged**